### PR TITLE
Migrate send-to-device widget to post directly to Basket (Fixes #11945)

### LINKED ISSUE #11945: Send to device forms should post directly to basket
## Description

Currently all send-to-device forms post to `/firefox/send-to-device-post/`. In an effort to lighten the load on bedrock, we'd like to switch these over to post to basket directly instead.

Endpoint: https://basket.mozilla.org/news/subscribe/
Docs: https://basket.readthedocs.io/newsletter_api.html#news-subscribe

---

#### Success Criteria
- [ ] Send to device forms post / interact directly with basket

### PR DESCRIPTION
**Pull request by alexgibson**
## One-line summary

Updates Send-to-Device widget to post directly to Basket from JS.

## Significant changes and points to review

- Converts`send_to_device` macro to a global Python helper for easier handling of message set logic / testing.
- Updates `send-to-device.js` to modern JS syntax.
- Changes default CSS class to `vertical`, as we don't use `horizontal` anywhere.
- I also removed `product` as a hidden form field / parameter, as this doesn't seem to be used anywhere by the backend code (as far as I can tell?). My guess is this may have been included originally for an option where the user can select a product to generate a link for, and has either since been removed or was never fully implemented?

## Issue / Bugzilla link

#11945

## Testing

- http://localhost:8000/en-US/firefox/welcome/4/
- http://localhost:8000/en-US/firefox/welcome/6/
- http://localhost:8000/en-US/firefox/browsers/mobile/?xv=legacy (test both forms at top / bottom)
- http://localhost:8000/en-US/firefox/browsers/mobile/android/
- http://localhost:8000/en-US/firefox/browsers/mobile/ios/
- http://localhost:8000/en-US/firefox/browsers/mobile/compare/
- http://localhost:8000/en-US/firefox/mobile/get-app/
- http://localhost:8000/en-US/firefox/whatsnew/
- http://localhost:8000/en-US/firefox/60.0/whatsnew/ (need to be signed into FxA)

Things to verify:

- [x] Email received OK for each URL / form
- [x] For locales such as `de` or `fr`, completing the form should generate a localised email.
- [x] Success and error messages displayed as expected (`success@example.com` and `failure@example.com`).
- [x] The custom SMS widget still works as expected: http://localhost:8000/en-US/firefox/browsers/mobile/android/?v=2 (use `5555555555` for success and `1234567899` for a failure).

Successful test run: https://gitlab.com/mozmeao/www-config/-/pipelines/623808523

### CODE REVIEW

File: bedrock/firefox/urls.py [hunk 1]
```diff
@@ -81,7 +81,6 @@
         ),
         name="firefox.mobile.get-app",
     ),
-    path("firefox/send-to-device-post/", views.send_to_device_ajax, name="firefox.send-to-device-post"),
```
Comment (alexgibson) [inline bedrock/firefox/urls.py:84]: Note to self: what should we do here, just let the URL 404?

Comment (alexgibson) [reply bedrock/firefox/urls.py:84]: @pmac interested in your opinion - what should we do with this URL, is it OK to just let this 404?

Comment (pmac) [reply bedrock/firefox/urls.py:84]: My opinion is yes, it should be fine for this to 404. We try not to do that with pages people may have visited, but an API endpoint like this that was just for our pages, let's just kill it. 

The `/newsletter/` endpoint may be another story as I know that there are other forms on other Mozilla sites that post there directly. We'll have to deal with that later and just leave it in place for now.

**Review by nathan-barrett — APPROVED** [review]
r+ Took a look over the FE code and nothing stood out to me needing to change. Excellent job with moving this all client side 💪🏻

**Review by stevejalim — APPROVED** [review]
Backendery looks good to me. Nice one! 🧺 🌟

**Pull request merged**